packages/editor/src/lib/editor/shapes/ShapeUtil.ts

```diff
@@ -32,6 +32,22 @@ export interface TLShapeUtilConstructor<
 /** @public */
 export type TLShapeUtilFlag<T> = (shape: T) => boolean
 
+/**@public */
+export interface TLShapeUtilCanBindOpts<Shape extends TLUnknownShape = TLShape> {
+	/** The shape from this util */
+	shape: Shape
+	/** The other shape being bound to. It may not exist yet, in which case it will be missing. */
+	otherShape?: TLShape
+	/**
+	 * The direction:
+	 * - `"to"` means the other shape is being bound to this shape.
+	 * - `"from"` means this shape is being bound to the other shape.
+	 */
+	direction: 'to' | 'from'
+	/** The type of binding, e.g. `"arrow"`. */
+	type: string
+}
+
```

Comment:
Something about shape / otherShape / direction had me confused at first, maybe its the framing of "this shape" vs "the other shape". `fromShape` / `toShape` seems more consistent with the binding format, though then it would be up to the user to figure out which shape (or botth) is the shape with the type corresponding to to this util. That might be ok.

If this function were a question, I think it would be "hey util, i'm trying to make a binding and at least one of these shapes is one of yours, does this look good to you?".